### PR TITLE
[GPU] Add i32 output type for FC

### PR DESCRIPTION
**Pull request by kelvinchoi-intel**
### Details:
 - *Add i32 output type for FC*

### Tickets:
 - *100911*

### CODE REVIEW

File: src/plugins/intel_gpu/tests/unit/test_cases/fully_connected_gpu_test.cpp [hunk 1]
```diff
@@ -316,6 +316,43 @@ TEST(fully_connected_gpu, no_biases_int8) {
...
+    network.set_input_data("input", input_prim);
+
+    auto outputs = network.execute();
+    ASSERT_EQ(outputs.size(), size_t(1));
+    ASSERT_EQ(outputs.begin()->first, "output");
+
+    auto output_prim = outputs.begin()->second.get_memory();
+    cldnn::mem_lock<ov::float16> output_ptr (output_prim, get_test_stream());
+    ASSERT_EQ(2, output_ptr[0]);
+}
```
[the remark this answers is not in the document]
Comment (kelvinchoi-intel) [reply src/plugins/intel_gpu/tests/unit/test_cases/fully_connected_gpu_test.cpp:354]: Added i32 type for FC func test.

**Review by hyunback — APPROVED** [review]

File: src/plugins/intel_gpu/tests/unit/test_cases/fully_connected_gpu_test.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (yeonbok) [reply src/plugins/intel_gpu/tests/unit/test_cases/fully_connected_gpu_test.cpp:354]: Umm.. what I meant why the test was using only such a tiny shape. it is [2,1][1,2]. Is there any reason?

[the remark this answers is not in the document]
Comment (yeonbok) [reply src/plugins/intel_gpu/tests/unit/test_cases/fully_connected_gpu_test.cpp:354]: Generally, to support a new feature we need to add a meaningful test to ensure the values too. To ensure meaningful test we need to ensure at least somewhat meaningful size of the tensor.

**Comment by yeonbok** [comment]
BTW I have a question, there is no i32 dtype in the implementation_map of fc layer. How can this be selected?

**Comment by kelvinchoi-intel** [comment]
> BTW I have a question, there is no i32 dtype in the implementation_map of fc layer. How can this be selected?
I've added only output data type. Does the implementation_map effect on output type?

**Pull request merged**